networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py

```diff
@@ -123,11 +126,19 @@ def get_vrf_context(self):
             return None
 
     def _get_vrf_context_gbp(self, context, router_id, port_db):
+        # TODO(tbachman): inefficient - should be fixed
         l3p_list = self.apic_driver.gbp_plugin.get_l3_policies(
-            context.elevated(), {'router': [router_id]})
-        kwargs = {'vrf_id': l3p_list[0]['id']}
+            context.elevated(), filters={'routers': [router_id]})
+        for l3p in l3p_list:
+            if l3p.get('routers') and router_id in l3p['routers']:
+                break
+        if not l3p_list or not l3p:
+            return None
+        kwargs = {'vrf_id': l3p['id']}
+        # TODO(tbachma): do we need to call get_vrf_details?
```

Comment:
If you already got the l3p ID then that's all you need. Don't need to call get_vrf_details() at all....